File: C-sharp-Web-Developer/C# Advanced/Sets and Dictionaries/05. Phone Book/PhoneBook.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

public class PhoneBook
{
    public static void Main()
    {
        var phonebook = new Dictionary<string, string>();
        var input = Console.ReadLine().Split(new char[] { '-' }, StringSplitOptions.RemoveEmptyEntries).ToArray();
        var result = new StringBuilder();

        while (input[0] != "search")
        {
            string name = input[0];
            string number = input[1];
            phonebook[name] = number;
            input = Console.ReadLine().Split(new char[] { '-' }, StringSplitOptions.RemoveEmptyEntries).ToArray();
        }

        var secondInput = Console.ReadLine().Split(new char[] { '-' }, StringSplitOptions.RemoveEmptyEntries).ToArray();

        while (!secondInput[0].ToLower().Equals("stop"))
        {
            if (phonebook.ContainsKey(secondInput[0]))
            {
                result.AppendLine($"{secondInput[0]} -> {phonebook[secondInput[0]]}");
            }
            else
            {
                result.AppendLine($"Contact {secondInput[0]} does not exist.");
            }
            secondInput = Console.ReadLine().Split(new char[] { '-' }, StringSplitOptions.RemoveEmptyEntries).ToArray();
        }
        Console.WriteLine(result.ToString());
    }
}

File: Software-Technologies/Java Exercises/Homework Part 3/src/StringsMapsStreamAPI/Phonebook.java
package StringsMapsStreamAPI;


import java.util.*;

public class Phonebook {

    public void phonebook() {
        Map<String, String> phonebook = new HashMap<>();

        boolean play = true;
        Scanner scanner = new Scanner(System.in);
        String[] input = new String[3];
        String name = "";
        String number = "";
        String command = "";

        while (play) {
            System.out.print("Please write your command: ");
            input = scanner.nextLine().split("\\s");
            command = input[0];

            if (command.equals("A")){
                name = input[1];
                number = input[2];
                phonebook.containsKey(name);
                phonebook.put(name, number);
            }
            else if (command.equals("S")){
                name = input[1];
                if (!phonebook.containsKey(name)){
                    System.out.printf("Contact %s does not exist. %n", name);
                }
                else {
                    System.out.printf("%s -> %s %n", name, phonebook.get(name));
                }
            }
            else if (command.equals("END")){
                play = false;
                break;
            }
        }
        scanner.close();
    }
}
